packages/app/src/state/staking/actions.ts

```diff
@@ -58,9 +58,43 @@ export const fetchStakingData = createAsyncThunk<
 	}
 })
 
+export const fetchStakingV2Data = createAsyncThunk<
+	{
+		rewardEscrowBalance: string
+		stakedNonEscrowedBalance: string
+		stakedEscrowedBalance: string
+		claimableBalance: string
+		totalStakedBalance: string
+		stakedResetTime: number
+		kwentaStakingV2Allowance: string
+	},
+	void,
+	ThunkConfig
+>('staking/fetchStakingDataV2', async (_, { extra: { sdk } }) => {
+	const {
+		rewardEscrowBalance,
+		stakedNonEscrowedBalance,
+		stakedEscrowedBalance,
+		claimableBalance,
+		totalStakedBalance,
+		stakedResetTime,
+		kwentaStakingV2Allowance,
+	} = await sdk.kwentaToken.getStakingV2Data()
+
```

Comment:
Would be good to wrap in try catch and notify on error, similar to the pattern used in futures.

Reply:
Yes. Good idea. Updated.